### ISSUE TITLE
Network structure issues and outcome issues

### ISSUE THREAD

**Issue by save-me-ctrl-c** [panyi95/DEFNet#3]
the shapel of out_put is [2,1,64,64], but need a [,,32,32] shape, so i add a conv at the end of reglayer in the Mymodule.py :   self.reg_layer = nn.Sequential(
            nn.Conv2d(160,64,kernel_size=3,stride=2,padding=1),
            nn.BatchNorm2d(64),
            nn.ReLU(),
            nn.Conv2d(64,32,kernel_size=3,stride=2,padding=1),
            nn.BatchNorm2d(32),
            nn.ReLU(),
            nn.Conv2d(32,16,1),
            nn.BatchNorm2d(16),
            nn.ReLU(),
            nn.Conv2d(16,1,kernel_size=1),
            nn.BatchNorm2d(1),
            nn.ReLU(),
            nn.Conv2d(1, 1, kernel_size=3, stride=2, padding=1),
            nn.BatchNorm2d(1),
            nn.ReLU()
            )
but i find that the result data has a significant difference. 
![Snipaste_2024-07-17_11-24-41](https://github.com/user-attachments/assets/859aa94f-f388-4cbd-b2f7-154c959c87e6)

Sincerely, I would like to ask you how the actual code completes the matching problem of data dimensions。

**Comment by sandipan211** [comment]
> the shapel of out_put is [2,1,64,64], but need a [,,32,32] shape, so i add a conv at the end of reglayer in the Mymodule.py : self.reg_layer = nn.Sequential( nn.Conv2d(160,64,kernel_size=3,stride=2,padding=1), nn.BatchNorm2d(64), nn.ReLU(), nn.Conv2d(64,32,kernel_size=3,stride=2,padding=1), nn.BatchNorm2d(32), nn.ReLU(), nn.Conv2d(32,16,1), nn.BatchNorm2d(16), nn.ReLU(), nn.Conv2d(16,1,kernel_size=1), nn.BatchNorm2d(1), nn.ReLU(), nn.Conv2d(1, 1, kernel_size=3, stride=2, padding=1), nn.BatchNorm2d(1), nn.ReLU() ) but i find that the result data has a significant difference. ![Snipaste_2024-07-17_11-24-41](https://github.com/user-attachments/assets/859aa94f-f388-4cbd-b2f7-154c959c87e6)
> 
> Sincerely, I would like to ask you how the actual code completes the matching problem of data dimensions。

hey @save-me-ctrl-c, I am facing a similar issue - could you find a solution to this?
